onnxruntime/test/providers/qnn/reduce_ops/reduce_op_cpu_test.cc

```diff
@@ -0,0 +1,435 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if !defined(ORT_MINIMAL_BUILD)
+
+#include <string>
+#include <unordered_map>
+
+#include "test/optimizer/qdq_test_utils.h"
+#include "test/providers/qnn/qnn_test_utils.h"
+
+#include "gtest/gtest.h"
+
+namespace onnxruntime {
+namespace test {
+
+/**
+ * Creates a graph with a single reduce operator (e.g., ReduceSum, ReduceMin, etc.). Reduce operators take the
+ * axes of reduction as either a node attribute or an optional input (depending on opset).
+ *
+ * \param reduce_op_type The string denoting the reduce operator's type (e.g., "ReduceSum").
+ * \param input_shape The shape of the input. Input data is randomly generated with this shape.
+ * \param axes_as_input True if the "axes" are specified as a node input.
+ * \param axes The axes of reduction.
+ * \param keepdims True if the output's rank should match the input. This is a node attribute that defaults to true.
+ * \param noop_with_empty_axes True if empty axes should force the node to act as a NoOp (no operation).
+ *                             This is a node attribute that defaults to false.
+ * \param domain The domain to assign to the graph node.
+ *
+ * \return A function that builds the graph with the provided builder.
+ */
+template <typename DataType>
+static GetQDQTestCaseFn BuildReduceOpTestCase(const std::string& reduce_op_type,
```

Comment:
>GetQDQTestCaseFn [](start = 7, length = 16)

Do these tests has anything to do with QDQ? Why we need to keep these Non-QDQ tests for CPU backend here for Qnn EP only? We should re-use the non-QDQ related tests with CPU EP in test/providers/cpu folder, so it can be reused for all EPs.

Reply:
These tests are for QnnCpu (not qdq). 

The tests in test/provider/cpu will pass as long as one EP can handle the test. So, if we introduce an error in the QNN EP, those tests will probably still pass because of the CPU EP.

These tests will always explicitly target QNN EP. They also allow us to check that all nodes are assigned to QNN, and that the model output matches CPU EP.